python/pyarrow/tests/test_table.py

```diff
@@ -3357,3 +3357,137 @@ def test_invalid_non_join_column():
     with pytest.raises(pa.lib.ArrowInvalid) as excinfo:
         t2.join(t1, 'id', join_type='inner')
     assert exp_error_msg in str(excinfo.value)
+
```

Comment:
I might move tests over to `test_cuda.py` and break them out into smaller test cases.

Reply:
Also, the test coverage can still be improved.